Blog - Deploy Applications to AWS ECR with a GitHub Actions CI/CD Pipeline (#404)
earthly/website: blog/_posts/2023-05-25-aws-ecr-github-actions.md

## Docker Deployments With AWS ECR

You require a packaged application that is easily portable throughout the pipeline. Docker provides a consistent and isolated environment for building, testing, and deploying applications. It allows you to create Docker images that package an application and its dependencies into a single portable artifact. This makes it easier to trigger changes across different stages of the pipeline.

To manage a [Docker](/blog/rails-with-docker) image, you need infrastructures allowing you to store and retrieve images while maintaining application scalability. AWS provides [ECR](https://aws.amazon.com/ecr/) as a fully managed container registry service to deploy your application using Docker.

Let's dive in and create a GitHub Actions workflow that will automate deployments to [AWS ECR](https://docs.aws.amazon.com/AmazonECR/latest/userguide/what-is-ecr.html).

### Prerequisites

> Before proceeding with this guide, you require the following:

Reviewer: ```suggestion
To follow along with this guide, you require the following:
```

              id: login-ecr
              uses: aws-actions/amazon-ecr-login@v1

            - name: Build, tag, and push image to Amazon ECR 
            # copying the code from repo i.e. Dockerfile, versioning 
            # the docker image, and pushing it to ECR.
              env:
                ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
                ECR_REPOSITORY: simple_nodejs_app
                IMAGE_TAG: nodejs_simple_app
              run: |
                docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
                docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
~~~

## Deploying to AWS ECR

To deploy your workflow, you only need to hit the `Start commit` as follows:

<div class="wide">
![Image guide]({{site.images}}{{page.slug}}/7vNrG0d.png)\
</div>

Once you have committed, a workflow should be started automatically.

<div class="wide">
![Image guide]({{site.images}}{{page.slug}}/XBwYXnx.png)\
</div>

Go ahead and refresh your ECR repository, and your application image will be deployed as follows:

<div class="wide">
![Image guide]({{site.images}}{{page.slug}}/jfi5YV3.png)\
</div>

## Conclusion

> 
> Deploying applications to AWS ECR with a GitHub Actions CI/CD creates a reliable pipeline that automates Docker builds and [deployment](/blog/deployment-strategies) cycles. This guide helped you learn how to deploy an application to AWS ECR using a GitHub Actions CI/CD pipeline. I hope you found the GitHub Actions workflow useful while leveraging automation to AWS resources. To further improve your [CI/CD pipelines](https://earthly.dev/blog/ci-vs-cd/), leverage other AWS services, such as [CodePipeline](https://aws.amazon.com/codedeploy/) and [CodeDeploy](https://aws.amazon.com/codepipeline/), and streamline your pipeline infrastructure.
> 

Reviewer: ```suggestion

Deploying applications to AWS ECR with a GitHub Actions CI/CD creates a reliable pipeline that automates Docker builds and [deployment](/blog/deployment-strategies) cycles. This guide helped you learn how to deploy an application to AWS ECR using a GitHub Actions CI/CD pipeline. I hope you found the GitHub Actions workflow useful while leveraging automation to AWS resources.

To further improve your [CI/CD pipelines](https://earthly.dev/blog/ci-vs-cd/), leverage other AWS services, such as [CodePipeline](https://aws.amazon.com/codedeploy/) and [CodeDeploy](https://aws.amazon.com/codepipeline/), and streamline your pipeline infrastructure.

```
